File: AZ-204_Implement_Azure_Functions/code/resource_mgmnt.cs
using System;
using Microsoft.Azure.Management.ResourceManager.Fluent;
using Microsoft.Azure.Management.ResourceManager.Fluent.Authentication;
using Microsoft.Azure.Management.ResourceManager.Fluent.Models;
using Microsoft.Azure.Management.ResourceManager.Fluent.Core;

class Program
{
    static void Main(string[] args)
    {
        // Authenticate using your Azure credentials
        var credentials = SdkContext.AzureCredentialsFactory.FromFile("azureauth.properties");

        // Create an Azure resource management client
        var azure = Azure.Configure()
            .WithLogLevel(HttpLoggingDelegatingHandler.Level.Basic)
            .Authenticate(credentials)
            .WithDefaultSubscription();

        // Provision an Azure Function
        var functionApp = azure.AppServices.FunctionApps.Define("myFunctionApp")
            .WithRegion(Region.USWest)
            .WithNewResourceGroup("myResourceGroup")
            .WithNewConsumptionPlan()
            .Create();

        Console.WriteLine("Azure Function provisioned successfully!");

        // List all Azure Functions in the subscription
        var functionApps = azure.AppServices.FunctionApps.List();
        foreach (var app in functionApps)
        {
            Console.WriteLine($"Azure Function: {app.Name}");
        }

        // Update the Azure Function
        functionApp.Update()
            .WithAppSetting("MySetting", "MyValue")
            .Apply();

        Console.WriteLine("Azure Function updated successfully!");

        // Delete the Azure Function
        azure.AppServices.FunctionApps.DeleteById(functionApp.Id);

        Console.WriteLine("Azure Function deleted successfully!");
    }
}

File: AZ-204_Implement_Azure_Functions/code/resource_mgmnt.py
from azure.identity import DefaultAzureCredential
from azure.mgmt.resource import ResourceManagementClient
from azure.mgmt.web import WebSiteManagementClient

# Set your Azure subscription ID and resource group name
subscription_id = '<your-subscription-id>'
resource_group_name = '<your-resource-group-name>'

# Set your Azure credentials
credential = DefaultAzureCredential()

# Create a resource management client
resource_client = ResourceManagementClient(credential, subscription_id)

# Create a web site management client
web_client = WebSiteManagementClient(credential, subscription_id)

# Provision an Azure Function
function_app_name = '<your-function-app-name>'
location = '<your-location>'
function_app = web_client.web_apps.create_or_update(resource_group_name, function_app_name, {
    'location': location,
    'kind': 'functionapp',
    'reserved': True,
    'site_config': {
        'app_settings': [
            {
                'name': 'FUNCTIONS_WORKER_RUNTIME',
                'value': 'python'
            }
        ]
    }
})

# List all Azure Functions in the resource group
functions = web_client.web_apps.list_by_resource_group(resource_group_name)
for function in functions:
    print(function.name)

# Update an Azure Function
function_app_name = '<your-function-app-name>'
function_name = '<your-function-name>'
function_app = web_client.web_apps.get(resource_group_name, function_app_name)
function_app.site_config.app_settings.append({
    'name': 'NEW_SETTING',
    'value': 'new-value'
})
web_client.web_apps.update(resource_group_name, function_app_name, function_app)

# Delete an Azure Function
function_app_name = '<your-function-app-name>'
web_client.web_apps.delete(resource_group_name, function_app_name)